Simplified/Accounts/AgeCheck/NYPLAgeCheck.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+
+protocol NYPLAgeCheckDelegate: class {
+  var minYear : Int { get }
+  var currentYear : Int { get }
+  var birthYearList : [Int] { get }
+  
+  func isValid(birthYear: Int) -> Bool
+  
+  func ageCheckCompleted(_ birthYear: Int)
+  func ageCheckFailed()
+}
+
+@objc protocol NYPLAgeCheckVerification {
+  func verifyCurrentAccountAgeRequirement(userAccountProvider: NYPLUserAccountProvider,
+                                          currentLibraryAccountProvider: NYPLCurrentLibraryAccountProvider,
+                                          _ completion: ((Bool) -> ())?) -> Void
+}
+
+@objcMembers final class NYPLAgeCheck : NSObject, NYPLAgeCheckDelegate, NYPLAgeCheckVerification {
```

Comment:
looking at this now the semantics of these 2 protocols are kind of unclear. 

`NYPLAgeCheckVerification` is the interface for what is actually determining the need for age verification. To be more swifty perhaps we should rename it to `NYPLAgeCheckVerifying`.

`NYPLAgeCheckDelegate` is the interface that defines the things required to perform age validation from the point of view of the component that has to enforce it (the VC in our case). So maybe we could rename it to `NYPLAgeCheckValidationDelegate`, since it's the VC who's delegating validation to `NYPLAgeCheck`?

Also curious to hear @singhraman4282 thoughts.

Reply:
I agree with @ettore here. I usually follow the swift style guide for protocol naming https://swift.org/documentation/api-design-guidelines/#protocols-describing-what-is-should-read-as-nouns. 

Protocols that describe a capability should be named using the suffixes able, ible, or ing. Therefore NYPLAgeCheckVerification should be changed to NYPLAgeCheckVerifying.

Also, I think NYPLAgeCheckDelegate can use the Should, Will, Did protocol pattern. So func ageCheckCompleted(_ birthYear: Int) can be changed to didCompleteAgeCheck(_ birthYear: Int) and ageCheckFailed() can be changed to didFailAgeCheck(_ error: Error?)

Reply:
Updated the naming of the functions